transcom/mymove
# I-12678 check payment requests table rather than payment service items table

## [Agility issue ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=I-12678)
## [Integration PR](https://github.com/transcom/mymove/pull/12238)

Needed to check the `payment_request` table for a status in addition to the `payment_service_item` status table. Now will check  payment_request table to see if payment request is in `reviewed`, `sent to gex`, `received by gex`, or `paid` states. Also checking that the payment service item is not in a `denied` state.

To test:
create payment request on a DLH or DSH, don't review as office user yet
go from shorthaul to linehaul with a destination address change
DSH should get rejected, DLH should be regenerated

approve payment request
short from linehaul to shorthaul with a destination address change
DLH should not get rejected
